File: Ex3/Ex3.go
package main

import (
	"fmt"
	"math"
)

// Функция для проверки, является ли число простым
func isPrime(num int) bool {
	if num <= 1 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(num))); i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

// Функция для решения задачи
func countPrimes(n int) int {
	count := 0
	for i := 0; i < n; i++ {
		var num int
		fmt.Print("Enter the number: ")
		fmt.Scan(&num)
		if isPrime(num) {
			count++
		}
	}
	fmt.Print("Number of prime numbers: ")
	return count
}

func main() {
	var n int
	fmt.Print("Enter the number of numbers: ")
	fmt.Scan(&n)
	fmt.Println(countPrimes(n))
}

File: Ex3/Ex3.hs
-- Функция для проверки, является ли число простым
isPrime :: Int -> Bool
isPrime num
    | num <= 1 = False
    | otherwise = all (\x -> num `mod` x /= 0) [2..floor . sqrt $ fromIntegral num]

-- Функция для решения задачи
countPrimes :: Int -> IO Int
countPrimes n = do
    numbers <- sequence [getNumber | _ <- [1..n]]
    let primeCount = length . filter isPrime $ numbers
    putStrLn $ "Number of prime numbers: " ++ show primeCount
    return primeCount

-- Функция для получения числа из пользовательского ввода
getNumber :: IO Int
getNumber = do
    putStrLn "Enter the number: "
    num <- readLn
    return num

main :: IO ()
main = do
    putStrLn "Enter the number of numbers: "
    n <- readLn
    countPrimes n
    return ()